src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -190,6 +190,12 @@ internal void CreateLuceneIndexWriter(OpenMode mode = OpenMode.CREATE)
             }
         }
 
+        private void InitializeIndexSearcher()
+        {
+            dirReader = writer != null ? writer.GetReader(applyAllDeletes: true) : DirectoryReader.Open(indexDir);
+            Searcher = new(dirReader);
+        }
```

Comment:
not sure if I should lock this, expecting concurrent access to `Searcher`
All the indexing changes will probably not happen while a user is actively searching ...